### PR TITLE
Fix Kinesis integration test

### PR DESCRIPTION
**Pull request by cbornet**
### Motivation
The current `PulsarSinksTest::testKinesis` is incorrect:
* If the `GetRecordsResponse` doesn't contain any record, it passes
* It configures the Sink to use the testcontainer mapped port instead of the internal one so the Sink can't connect to Localstack

### Modifications
* Use port 4566 for `awsEndpointPort` which is the unmapped Localstack service port
* Verify that all records received match exactly the one sent

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:
* Run test `PulsarSinksTest::testKinesis`

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no) no
  - The public API: (yes / no) no
  - The schema: (yes / no / don't know) no
  - The default values of configurations: (yes / no) no
  - The wire protocol: (yes / no) no
  - The rest endpoints: (yes / no) no
  - The admin cli options: (yes / no) no
  - Anything that affects deployment: (yes / no / don't know) no

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why) it's a test
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
Great catch !

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java [hunk 1]
```diff
@@ -64,16 +63,11 @@ public void prepareSink() throws Exception {
...
         sinkConfig.put("skipCertificateValidation", true);
-        client = KinesisAsyncClient.builder().credentialsProvider(new AwsCredentialsProvider() {
-                    @Override
-                    public AwsCredentials resolveCredentials() {
-                        return AwsBasicCredentials.create(
-                                "access",
-                                "secret");
-                    }
-                })
+        client = KinesisAsyncClient.builder().credentialsProvider(() -> AwsBasicCredentials.create(
```
Comment (nicoloboschi) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:68]: do we close this client?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
good catch! 
I observed now that we don't close the KinesisClient in this test, can you fix it?

**Comment by cbornet** [comment]
True but there's currently no method to do that in `SinkTester`

**Comment by nicoloboschi** [comment]
I can do it today. I'm ok to merge this pr

**Review by nicoloboschi — APPROVED** [review]

**Comment by cbornet** [comment]
I had missed that there's a `stopServiceContainer` method that can be overridden. I'll do it.

**Review by lhotari — APPROVED** [review]

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java [hunk 2]
```diff
@@ -18,35 +18,34 @@
...
 import java.util.Map;
 
-import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
 
 @Slf4j
 public class KinesisSinkTester extends SinkTester<LocalStackContainer> {
 
     private static final String NAME = "kinesis";
+    private static final int LOCALSTACK_SERVICE_PORT = 4566;
```
Comment (lhotari) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:48]: Just wondering about the fixed port number. Usually Testcontainers uses dynamic ports for everything so I'd assume that also in this case.

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java [hunk 3]
```diff
@@ -64,16 +63,11 @@ public void prepareSink() throws Exception {
         final LocalStackContainer localStackContainer = getServiceContainer();
         final URI endpointOverride = localStackContainer.getEndpointOverride(LocalStackContainer.Service.KINESIS);
         sinkConfig.put("awsEndpoint", NAME);
-        sinkConfig.put("awsEndpointPort", endpointOverride.getPort());
+        sinkConfig.put("awsEndpointPort", LOCALSTACK_SERVICE_PORT);
```
Comment (lhotari) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:66]: Is this change necessary?

**Review by lhotari — CHANGES_REQUESTED** [review]
please check the questions about the fixed port

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java (hunk 3 above)
Comment (eolivelli) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:66]: it is the core of the problem :-)

Comment (cbornet) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:66]: yes it is.
We were using the port forwarded to the host. But the sink is not on the host but in a container so it must use the internal container port.

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java (hunk 2 above)
Comment (cbornet) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:48]: It is this port : https://github.com/testcontainers/testcontainers-java/blob/0408db806f6d57b8117881adcd10b6bd6c9dcd9c/modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java#L38
I didn't find an easy way to get it from Testcontainers.
It's not dynamic as we use the internal container port

Comment (lhotari) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:48]: makes sense

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java (hunk 3 above)
Comment (lhotari) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:66]: ok makes sense. Thanks for explaining.

**Review by lhotari — APPROVED** [review]
LGTM

**Review by dlg99 — APPROVED** [review]

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java (hunk 1 above)
Comment (cbornet) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java:68]: done
